### DISCUSSION TITLE
Help getting LILYGO TTGO LoRa32 V2.1 loaded with firmware

### DISCUSSION THREAD

**Discussion by silveroranges** [markqvist/Reticulum#60] [Help & Questions] [answered]
So I hope I got the right board from aliexpress since it took like 3 months to get here haha. 

I have had the runaround, I hadn't loaded ubuntu in my vmware for months and for some reason it wouldn't update. Did a fresh install, got the rnodeconf installed, I am trying to do an automated setup. I entered all the correct hardware info, and then it tells me that python isn't found in usr/bin/env. I know its probably something super obvious. I have the board connected via USB, I didn't install any drivers for it. 

I saw the discussion where the german fellow has the same [issue ](https://github.com/markqvist/rnodeconfigutil/issues/13) I tried doing the workaround using alias python=python3 and ran the autoinstaller again but got the same issue. 

I haven't installed anything onto the board yet, no SD card, etc. Figured I'd make a discussion to help future noobs out. Only other thing I can think of is that since it is a virtual machine it doesn't wanna do it. I'll break out a rpi tomorrow and see if that is the proble if I don't figure it out by then :)

![image](https://user-images.githubusercontent.com/22848776/170384596-2bf78a01-3869-4c31-9316-3236f81e061f.png)

**Comment by 4c3e** [comment, marked as answer]
Hello! There's a chance you're running into a common issue with ubuntu, can you try following the responses here: https://stackoverflow.com/questions/3655306/ubuntu-usr-bin-env-python-no-such-file-or-directory

**Reply by silveroranges** [reply]
Awesome, that stack overflow thread fixed that problem. Now I am getting a timed out error for the board, I didn't have time to do anything except run the auto load script before I had to go to work. I will update this comment with a screenshot when I get home. It did seem to connect to the board, but i kind of looked like the board just reset instantly. I have a suspicion of what the problem is though. Either way I will update this when I get home. 

EDIT: Yeah, it didn't want to connect. I think it was because it was in a VM. I just installed the rpi os on my rpi4, the instructions on the repository worked perfectly first time. Loaded the firmware perfectly on all of the LilyGos.

**Reply by 4c3e** [reply]
Nice! Glad it worked out in the end :smile:

**Comment by markqvist** [comment]
The issue occurs because there is no clear consensus about the naming of the Python 3 executable across OSes. The most widely compatible right now seems to be to use "python3" though, and `esp-tool` uses "python", which doesn't work on some OSes, before explicitly stating that "python" is really equal to "python3".

I should probably just modify the `esp-tool` that rnodeconfig uses to just use "python3" to avoid this problem.
